bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterBridgeHandler.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meater.internal.handler;
+
+import static org.openhab.binding.meater.internal.MeaterBindingConstants.THING_TYPE_BRIDGE;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.meater.internal.MeaterBridgeConfiguration;
+import org.openhab.binding.meater.internal.api.MeaterRestAPI;
+import org.openhab.binding.meater.internal.discovery.MeaterDiscoveryService;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MeaterBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterBridgeHandler extends BaseBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private int refreshTimeInSeconds = 300;
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private LocaleProvider localeProvider;
+    private final Map<String, MeaterProbeDTO.Device> meaterProbeThings = new ConcurrentHashMap<>();
+
+    private @Nullable MeaterRestAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public MeaterBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson, LocaleProvider localeProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+        this.localeProvider = localeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        MeaterBridgeConfiguration config = getConfigAs(MeaterBridgeConfiguration.class);
+
+        MeaterRestAPI meaterRestAPI = new MeaterRestAPI(config, gson, httpClient,
+                localeProvider.getLocale().getLanguage());
```

Comment:
You will need to pass `LocaleProvider` to `MeaterRestAPI` so it can pick up changes in configured language after initialization.

Reply:
I have done that.

Reply:
Currently language is passed, not `LocaleProvider`. So the MeaterRestAPI object will still be using the language from when the object was constructed even after the user has changed openHAB language.

Reply:
I'll change that.